### PR TITLE
feat: yapf - added command into tox file to format code

### LINKED ISSUE #102: feat: integrating yapf into the armada
We would like to add the yapf tool into armada in order to test and format
**Is this a bug report or feature request?** (choose one): feat

**Python Version** (output of `python --version`): 2.7

**Development or Deployment Environment?**: Develop

**Release Tag or Master**: Master

**Expected Behavior**: 
- [ ] create a tox command to format the current source code
- [ ] create a tox command to validate that the code has been correctly formatted to be used for testing.

**Any Additional Comments**:

https://github.com/google/yapf

### PR DESCRIPTION
**Pull request by One-Fine-Day**
-updated tox
-added check_diff.sh

**What is the purpose of this pull request?**: feat improve lint

**What issue does this pull request address?**: Fixes #102

**Notes for reviewers to consider**:

**Specific reviewers for pull request**:
@gardlt @mark-burnett

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -10,7 +11,7 @@ setenv=
 
 [testenv:ubuntu]
 commands =
-	 sh {toxinidir}/tools/libgit2.sh
+         sh {toxinidir}/tools/libgit2.sh
```
Comment (gardlt) [inline tox.ini]: can you bring this up to L13

File: tox.ini [hunk 2]
```diff
@@ -19,12 +20,28 @@ commands = python setup.py build_sphinx
 commands = oslo-config-generator --config-file=etc/armada/config-generator.conf
 
 [testenv:lint]
-commands = flake8 .
+commands = flake8
+ignore=E302,H306,E402,W503
```
Comment (gardlt) [inline tox.ini]: i know that we used this test, but we should figure out a better way to approach this

File: tox.ini [hunk 3]
```diff
@@ -19,12 +20,28 @@ commands = python setup.py build_sphinx
...
+exclude= venv, .venv, libgit2-0.25.0, .git, .idea, .tox, *.egg-info, *.eggs, bin, dist, hapi
 
 [testenv:testing]
 commands = nosetest -w armada
 
+[testenv:checkdiff]
+deps = yapf==0.16.3
```
Comment (gardlt) [inline tox.ini]: lets add yapf into the `test-requirements.txt`

File: tox.ini (hunk 1 above)
Comment (1FineDay) [inline tox.ini]: Done!

File: tox.ini (hunk 3 above)
Comment (1FineDay) [inline tox.ini]: Done!

**Pull request closed without merging**